procedures/CodeBrowser_utils.ipf

```diff
@@ -166,91 +175,134 @@ End
+
+	myVar = numValue
+	if(numtype(numValue) == 0 && getGlobalVar(globalVar) != numValue)
+		// @todo There should be a test case for this
+		Abort "global variable content missmatch."
+	endif
+
+	return 1
 End
 
 // returns the Value of a (positive) numeric global Variable. Returns -1 if Variable does not exist.
+//
+// note: do not use `DebugPrint()` here, as it will cause recursion
+//
 Function getGlobalVar(globalVar)
 	String globalVar
-	DFREF dfr = createDFWithAllParents(pkgFolder)
 
-	NVAR/Z/SDFR=dfr myVar = dfr:$globalVar
+	DFREF dfr = $pkgFolder
+	if(DataFolderRefStatus(dfr) != 1)
+		return -1
+	endif
 
+	NVAR/Z/SDFR=dfr myVar = dfr:$globalVar
 	if(!NVAR_Exists(myVar))
 		return -1
-	else
-		return myVar
 	endif
+
+	return myVar
 End
 
 // set a global string variable
 Function setGlobalStr(globalVar, strValue)
 	String globalVar, strValue
-	DFREF dfr = createDFWithAllParents(pkgFolder)
 
+	DFREF dfr = createDFWithAllParents(pkgFolder)
 	String/G dfr:$globalVar
 	SVAR/Z/SDFR=dfr myVar = dfr:$globalVar
-
 	if(!SVAR_Exists(myVar))
-		DebugPrint("global String " + globalVar + " failed to create")
+		DebugPrint("failed to create global string " + globalVar)
 		return 0
-	else
-		myVar = strValue
-		return 1
 	endif
+
+	myVar = strValue
+	if(cmpstr(getGlobalStr(globalVar), strValue))
+		// @todo quick check! There should be a test case for this
+		Abort "global string content missmatch."
+		return 0
+	endif
+
+	return 1
 End
 
 // returns the Value of a global String. Returns NullString on Error.
 Function/S getGlobalStr(globalVar)
 	String globalVar
-	DFREF dfr = createDFWithAllParents(pkgFolder)
 
-	SVAR/Z/SDFR=dfr myVar = dfr:$globalVar
+	DFREF dfr = $pkgFolder
+	if(DataFolderRefStatus(dfr) != 1)
+		DebugPrint("Package DataFolder " + pkgFolder + " does not exist")
+		return ""
+	endif
 
+	SVAR/Z/SDFR=dfr myVar = dfr:$globalVar
 	if(!SVAR_Exists(myVar))
+		DebugPrint("global String does not exist")
 		return ""
-	else
-		return myVar
 	endif
+
+	return myVar
 End
 
+// @returns 1 if global string is not present
 Function killGlobalStr(globalVar)
 	String globalVar
-	DFREF dfr = createDFWithAllParents(pkgFolder)
-	SVAR/Z/SDFR=dfr myVar = dfr:$globalVar
 
-	KillStrings/Z dfr:$globalVar
+	DFREF dfr = $pkgFolder
+	if(DataFolderRefStatus(dfr) != 1)
+		DebugPrint("Package DataFolder " + pkgFolder + " does not exist", loglevel=3)
+		return 1
+	endif
 
+	SVAR/Z/SDFR=dfr myVar = dfr:$globalVar
 	if(!SVAR_Exists(myVar))
+		DebugPrint("Global String does not exist: " + globalVar, loglevel=3)
 		return 1
-	else
+	endif
+
+	KillStrings/Z dfr:$globalVar
+	SVAR/Z/SDFR=dfr myVar = dfr:$globalVar
+	if(SVAR_Exists(myVar))
+		DebugPrint("Could not delete global String " + globalVar)
 		return 0
 	endif
+
+	return 1
 End
 
 Function killGlobalVar(globalVar)
 	String globalVar
-	DFREF dfr = createDFWithAllParents(pkgFolder)
-	NVAR/Z/SDFR=dfr myVar = dfr:$globalVar
 
-	KillVariables/Z dfr:$globalVar
+	DFREF dfr = $pkgFolder
+	if(DataFolderRefStatus(dfr) != 1)
```

Comment:
Please use https://www.wavemetrics.com/code-snippet/datafolderexists-data-folder-references instead which has better semantics and works for free datafolders as well.